app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> None:
+    list_of_results = {}
+
+    def inner(*args: Any, **kwargs: Any) -> None:
+        if isinstance(args, (int, str, float, bool, tuple)):
+            if args not in list_of_results:
+                print("Calculating new result")
+                list_of_results[args] = func(*args)
+                return list_of_results[args]
```

Comment:
add else statement and remove duplicate of return list_of_results[args]

Reply:
I can't remove the duplicate "list_of_results[args]" because my tests give an error. only with 2 "return" ok.

Reply:
No it is false
You can or pin screenshot)
Because all tasks works without duplicate
You must remove line 12 and move line 15 to one left